File: java/739_DailyTemperatures.java
class Solution {
    public int[] dailyTemperatures(int[] T) {
        Stack<Integer> s = new Stack<>();
        int[] res = new int[T.length];
        for(int i = 0; i < T.length; i++) {
            while (!s.empty() && T[s.peek()] < T[i]) {
                int top = s.pop();
                res[top] = i - top;
            }
            s.push(i);
        }
        return res;
    }
}

File: c++/739_DailyTemperatures.cpp
class Solution {
public:
    vector<int> dailyTemperatures(vector<int>& t) {
        stack<int> stk;
        vector<int> res(t.size());
        for (int i = 0; i < t.size(); i++) {
            while (!stk.empty() && t[i] > t[stk.top()]) {
                int idx = stk.top();
                res[idx] = i - idx;
                stk.pop();
            }
            stk.push(i);
        }
        return res;
    }
};
